### ISSUE TITLE
False positive for snake_case lint when using bitflags macro

### ISSUE THREAD

**Issue by crumblingstatue** [rust-lang/rust-analyzer#8225]
```Rust
#[macro_use]
extern crate bitflags;

bitflags! {
    struct Flags: u32 {
        const A = 0;
    }
}
```
![image](https://user-images.githubusercontent.com/1521976/112749331-fd88e780-8fc1-11eb-996c-a37de3341eb4.png)

This seems to be the relevant expanded code
```Rust
#[allow(non_snake_case)]
        trait __BitFlags {
            #[inline]
            fn A(&self) -> bool {
                false
            }
        }
        impl __BitFlags for Flags {
            #[allow(deprecated)]
            #[inline]
            fn A(&self) -> bool {
                if Self::A.bits == 0 && self.bits != 0 {
                    false
                } else {
                    self.bits & Self::A.bits == Self::A.bits
                }
            }
        }
```
Rust-analyzer seems to ignore the `#[allow(non_snake_case)]` attribute. Rustc and clippy don't warn about this.

RA version is 2021-03-22

**Comment by Veykril** [comment]
Looks like we are case checking names in trait impls which isn't necessary, meanwhile we don't seem to check cases in trait definitions.

**Comment by PonasKovas** [comment]
Any progress?
